### PR TITLE
(665) Ingesting updates existing projects before creating new ones

### PR DESCRIPTION
**Pull request by tahb**
## Changes in this PR

- Update existing UKSA production data with the associated IATI identifiers to provide a reliable way to update existing records
- Never ingest the same activity twice to avoid double entry
- If the ingest finds an existing activity that matches the IATI activity, it will do a smaller set of actions. Namely creating transactions, budgets and planned descriptions, as well as saving the XML onto the record for future updates
- if the ingest cannot find a matching record (as is the case for 11 activities) then it will seek to create new records

### After

Screenshot 2020-05-21 at 17 59 10

Screenshot 2020-05-22 at 10 47 40
Screenshot 2020-05-22 at 10 47 47

Screenshot 2020-05-21 at 18 01 00

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

**Comment by tahb** [comment]
Requires https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/pull/411 to be merged first.

File: app/services/ingest_iati_activities.rb [hunk 1]
```diff
@@ -154,14 +158,14 @@ def call
       providing_organisation_reference = transaction_element.children.detect { |child| child.name.eql?("provider-org") }.attributes["ref"].value
       receiving_organisation_name = transaction_element.children.detect { |child| child.name.eql?("receiver-org") }.children.detect { |child| child.name.eql?("narrative") }.text
       receiving_organisation_reference = transaction_element.children.detect { |child| child.name.eql?("receiver-org") }.attributes["ref"].try(value, nil)
-
+# binding.pry
```
Comment (lozette) [inline app/services/ingest_iati_activities.rb]: 😱

Comment (tahb) [reply app/services/ingest_iati_activities.rb]: 🤭

Comment (tahb) [reply app/services/ingest_iati_activities.rb]: @lozette I've removed the pry, thanks for spotting.

**Pull request merged**